go/libraries/doltcore/sqle/dtables/diff_table.go

```diff
@@ -902,26 +902,31 @@ func GetDiffTableSchemaAndJoiner(format *types.NomsBinFormat, fromSch, toSch sch
 }
 
 // CalculateDiffSchema returns the schema for the dolt_diff table based on the schemas from the from and to tables.
-// Either may be nil, in which case it will be missing from the resulting schema.
+// Either may be nil, in which case the nil argument will use the schema of the non-nil argument
 func CalculateDiffSchema(fromSch, toSch schema.Schema) (schema.Schema, error) {
-
-	colCollection := schema.NewColCollection()
-	if fromSch != nil {
-		colCollection = fromSch.GetAllCols()
+	var toClmCol, fromClmCol *schema.ColCollection
+	if fromSch == nil && toSch == nil {
+		panic("non-nil argument required to CalculateDiffSchema")
```

Comment:
Avoid using `panic` and prefer just returning an error. Because `panic` will kill the entire go process, it should only be used if there is an extremely serious problem (e.g. data integrity risk) that would warrant stopping everything immediately. We want to protect a sql-server process used by multiple users from being killed abruptly just because one user hit a bug we weren't expecting. 

It this case, it does seem like this shouldn't ever happen, but returning an error instead will still have the effect of erroring out the query that caused the problem, without taking down the whole server.

Reply:
yep. facepalm